client/state/sites/plans/connections.js

```diff
@@ -0,0 +1,59 @@
+import debugFactory from 'debug';
+import map from 'lodash/map';
+
+import {
+	SITE_PLANS_FETCH_FAILED,
+	SITE_PLANS_TRIAL_CANCEL_COMPLETED,
+	SITE_PLANS_TRIAL_CANCEL_FAILED
+} from 'state/action-types';
+import { createSitePlanObject } from './assembler';
+import { fetchSitePlansCompleted } from './actions';
+import i18n from 'i18n-calypso';
+
+const debug = debugFactory( 'calypso:site-plans:connections' );
+
+export const connectCancelSitePlanTrial = wpcom => ( action, dispatch ) => {
+	return new Promise( ( resolve, reject ) => {
+		wpcom.undocumented().cancelPlanTrial( action.planId, ( error, data ) => {
+			if ( data && data.success ) {
+				dispatch( {
+					type: SITE_PLANS_TRIAL_CANCEL_COMPLETED,
+					siteId: action.siteId,
+					plans: map( data.plans, createSitePlanObject )
+				} );
+
+				resolve();
+			} else {
+				debug( 'Canceling site plan trial failed: ', error );
+
+				const errorMessage = error.message || i18n.translate( 'There was a problem canceling the plan trial. Please try again later or contact support.' );
+
+				dispatch( {
+					type: SITE_PLANS_TRIAL_CANCEL_FAILED,
+					siteId: action.siteId,
+					error: errorMessage
+				} );
+
+				reject( errorMessage );
+			}
+		} );
+	} );
+}
+
+export const connectFetchSitePlans = wpcom => ( action, dispatch ) =>
```

Comment:
I would also consider passing a callback function as a param instead of `wpcom` object, like:
```js
export const connectFetchSitePlans = getSitePlans => ( action, dispatch ) =>
    getSitePlans( action.siteId, ( error, data ) => {
    ...
```

First of all it would make unit testing even easier. You wouldn't have to know what structure `wpcom` has to write your test. I can imagine it would be also possible to pass several callbacks if there would be a need to cover a more complicated use case. Like when we need to execute several parallel async operations and dispatch SUCCESS  only when all of them are resolved.